modelforge/potential/parameters.py

```diff
@@ -81,6 +81,30 @@ class ActivationFunctionParamsEnum(CaseInsensitiveEnum):
     ELU = ActivationFunctionParamsAlpha
 
 
+class PredictedPropertiesMixin:
+    predicted_properties: List[str]
+    predicted_dim: List[int]
+
+    # Custom validation for handling the predicted_properties dictionary
+    @model_validator(mode="before")
+    def handle_predicted_properties_dict(cls, values: dict) -> dict:
```

Comment:
I don't think this routine is necessary at all. pydantic will automatically populate these fields.

Reply:
I didn't do a good job explaining what this does, but it is in fact necessary.

The input toml has the following structure

```
[potential.core_parameter.prediction]
predicted_properties = ["per_atom_energy"]
predicted_dim = [1]
```

that get's translated into a dictionary for which prediction is the key.

But the pydantic model wants `predicted_properties` and `predicted_dim` as keys (not inside a dictionary). To achieve this I need to 'flatten' the dictionary here.

Reply:
Would it just be easier to move this out of its own class at just put ‘predicted_properties’ and ‘predict_dims’ in CoreNetwork? We can still do model validation easily but the. They’d be properties the of core network and not prediction.